transforms/angle-brackets/transform.test.js

```diff
@@ -144,6 +144,68 @@ test('data-attributes', () => {
   `);
 });
 
+test('data-attributes-valueless', () => {
+  let options = {
+    includeValuelessDataAttributes: true,
+  };
+  let input = `
+    {{x-foo data-foo=true}}
+    {{x-foo data-test-selector=true}}
+    {{x-foo data-test-selector=post.id}}
+    {{x-foo label="hi" data-test-selector=true}}
+    {{x-foo data-test-foo }}
+
+    {{#x-foo data-foo=true}}
+      block
+    {{/x-foo}}
+
+    {{#x-foo data-test-selector=true}}
+      block
+    {{/x-foo}}
+
+    {{#x-foo data-test-selector=post.id}}
+      block
+    {{/x-foo}}
+
+    {{#common/accordion-component data-test-accordion as |accordion|}}
+      block
+    {{/common/accordion-component}}
+
+    {{x-foo
+      data-foo
+      name="Sophie"
+    }}
+  `;
+
+  expect(runTest('data-attributes-valueless.hbs', input, options)).toMatchInlineSnapshot(`
+    "
+        <XFoo @data-foo={{true}} />
+        <XFoo @data-test-selector={{true}} />
+        <XFoo @data-test-selector={{post.id}} />
+        <XFoo @label=\\"hi\\" @data-test-selector={{true}} />
+        <XFoo data-test-foo />
+
+        <XFoo @data-foo={{true}}>
+          block
+        </XFoo>
+
+        <XFoo @data-test-selector={{true}}>
+          block
+        </XFoo>
+
+        <XFoo @data-test-selector={{post.id}}>
+          block
+        </XFoo>
+
+        <Common::AccordionComponent data-test-accordion as |accordion|>
+          block
+        </Common::AccordionComponent>
+
+        <XFoo data-foo @name=\\"Sophie\\" />
```

Comment:
This transformation is not correct since the component itself will not see `data-foo` as a string. It would be whatever value `data-foo` references to.
`data-test` is special because of `ember-test-selectors`, and hence we should only assume this logic for `data-test` selectors and not for others.
@rwjblue  thoughts?

Reply:
a good example is here
```
{{#link-to
      "cart"
      data-test-foo
      class="nav-actions__cart link--wo-style"
    }}
```
Will render as: 

```
<a data-test-foo href="/cart" id="ember183" class="nav-actions__cart link--wo-style ember-view">      
</a>
```

whereas:

```
{{#link-to
      "cart"
      data-foo
      class="nav-actions__cart link--wo-style"
    }}
```
would output:
```
<a href="#" id="ember183" class="nav-actions__cart link--wo-style loading ember-view">      
 </a>
```

Reply:
That makes sense. I've updated my PR so it only transforms data-test attributes.